wordpress-mobile/WordPress-iOS
# Task/16941 - Close account

Fixes [#16941](https://github.com/wordpress-mobile/WordPress-iOS/issues/16941)
WPKit: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/435



This PR includes UI + logic for closing user account.

## Note
- We rely on API to [determine](https://github.com/Automattic/wp-calypso/blob/b550e845f46f9c93641f54193d73626cb5b669e9/client/me/account-close/main.jsx#L90) if account deletion is possible

## Tasks
- [x] Implement design
- [x] Implement close account logic
- [x] Error handling in case account deletion is not possible

## To test the happy path:
1. Create a temporary user account that will be deleted
2. Use the newly created account to log in to the app
3. Under **My Site** tab tap on profile button at the top right corner
4. Tap on **Account Settings**
5. Tap on **Close account**
6. Close account alert is displayed
7. Confirm close account by entering your username
7. Tap on **Permanently Close Account**
8. After the account has been closed, user is redirected the the welcome screen

## Happy path flow
https://user-images.githubusercontent.com/12729242/129917471-e974745d-efec-425d-bab1-d747b3c340b8.mp4

## To test the error:
1. Create a temporary user account that will be deleted
2. Create a test site using newly created user
3. Add a yearly Premium plan subscription to test site
4. Follow the steps 2-7 from the happy path
5. Error message is displayed

## Error alert
<img width="509" alt="Screen Shot 2021-08-19 at 10 18 52" src="https://user-images.githubusercontent.com/12729242/130040859-e66728b2-4d3f-416a-a939-686ece18f5a2.png">

## Regression Notes
1. Potential unintended areas of impact
n/a

2. What I did to test those areas of impact (or what existing automated tests I relied on)
n/a

3. What automated tests I added (or what prevented me from doing so)
n/a

PR submission checklist:
